nautobot/extras/models/groups.py

```diff
@@ -17,16 +18,152 @@
+
+            if self.content_type != database_object.content_type:
+                raise ValidationError({"content_type": "ContentType cannot be changed once created"})
+
+    def _check_valid_member(self, member):
+        if not (isinstance(member, self.model) or issubclass(member, self.model)):
+            raise TypeError(f"{member} is not a {self.model._meta.label_lower}")
+
+    def add_members(self, objects_to_add):
+        """Add the given list or QuerySet of objects to this Static Group."""
+        if isinstance(objects_to_add, models.QuerySet):
+            self._check_valid_member(objects_to_add.model)
+        else:
+            for obj in objects_to_add:
+                self._check_valid_member(obj)
+
+        for obj in objects_to_add:
+            # We don't use `.bulk_create()` currently because we want change logging for these creates.
+            # Might be a good future performance improvement though.
+            StaticGroupAssociation.objects.get_or_create(
+                static_group=self, associated_object_type=self.content_type, associated_object_id=obj.pk
+            )
+
+    def remove_members(self, objects_to_remove):
+        """Remove the given list or QuerySet of objects from this Static Group."""
+        if isinstance(objects_to_remove, models.QuerySet):
+            self._check_valid_member(objects_to_remove.model)
+            StaticGroupAssociation.objects.filter(
+                static_group=self,
+                associated_object_type=self.content_type,
+                associated_object_id__in=objects_to_remove.values_list("pk", flat=True),
+            ).delete()
+        else:
+            pks_to_remove = set()
+            for obj in objects_to_remove:
+                self._check_valid_member(obj)
+                pks_to_remove.add(obj.pk)
+
+            StaticGroupAssociation.objects.filter(
+                static_group=self, associated_object_type=self.content_type, associated_object_id__in=pks_to_remove
+            ).delete()
+
+
+@extras_features(
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class StaticGroupAssociation(OrganizationalModel):
+    """Intermediary model for associating an object to a StaticGroup."""
+
+    static_group = models.ForeignKey(to=StaticGroup, on_delete=models.CASCADE, related_name="static_group_associations")
+    associated_object_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        related_name="static_group_associations",
+        limit_choices_to=FeatureQuery("static_groups"),
+    )
+    associated_object_id = models.UUIDField(db_index=True)
+    associated_object = GenericForeignKey(ct_field="associated_object_type", fk_field="associated_object_id")
+
+    is_contact_associable_model = False
+    is_static_group_associable_model = False
+
+    class Meta:
+        unique_together = ("static_group", "associated_object_type", "associated_object_id")
```

Comment:
Should probably at least add `ordering = ["static_group"]` even if we can't define a more specific ordering?